nsqd/protocol_v2.go

```diff
@@ -712,9 +712,10 @@ func (p *protocolV2) REQ(client *clientV2, params [][]byte) ([]byte, error) {
 	}
 	timeoutDuration := time.Duration(timeoutMs) * time.Millisecond
 
-	if timeoutDuration < 0 || timeoutDuration > p.ctx.nsqd.getOpts().MaxReqTimeout {
-		return nil, protocol.NewFatalClientErr(nil, "E_INVALID",
```

Comment:
@mreiferson I think hanging up at `IDENTIFY` and `REQ` is worth thinking about some more. We actively monitor client errors, whereas early de-queues are not something we'd figure out right away. I would be happy with "`nsqd` requeued your message, but not exactly as you requested" but there's no mechanism for that. Our use case is we have some last ditch effort `REQ`'s in the `12h` range, if `nsqd` was configured incorrectly we'd see these come around again in an hour. Something to consider.